### PR TITLE
C++: More QLDoc

### PR DESCRIPTION
**Pull request by geoffw0**
Add more QLDoc in various places.  Deprecate `Member.qll` and some overly specific stuff in `Type.qll` (none of these things are meaningfully used as far as I can tell).

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/Type.qll [hunk 1]
```diff
@@ -1,5 +1,8 @@
+/**
+ * Provides a hierarchy of classes for modelling C/C++ types.
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Type.qll:2]:
```suggestion
 * Provides a hierarchy of classes for modeling C/C++ types.
```

File: cpp/ql/src/semmle/code/cpp/stmts/Stmt.qll [hunk 2]
```diff
@@ -1,3 +1,7 @@
+/**
+ * Provides a hierarchy of classes for modelling C/C++ statements.
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/stmts/Stmt.qll:2]:
```suggestion
 * Provides a hierarchy of classes for modeling C/C++ statements.
```

File: cpp/ql/src/semmle/code/cpp/Namespace.qll [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/**
+ * Provides classes for modelling namespaces, `using` directives and `using` declarations.
```
[the remark this answers is not in the document]
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Namespace.qll:2]: There are other uses of "modelling" in our code, I should make another PR to fix them elsewhere.

**Comment by geoffw0** [comment]
Fixed issues.

File: cpp/ql/src/semmle/code/cpp/Namespace.qll (hunk 3 above)
[the remark this answers is not in the document]
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Namespace.qll]: I've accepted your fixes, changes to other uses are here: https://github.com/github/codeql/pull/3802

**Pull request merged**